tests/e2e/customsuites/multivolume.go

```diff
@@ -0,0 +1,306 @@
+// - Multiple pods accessing the same S3 volume simultaneously
+// - A single pod accessing multiple S3 volumes concurrently
+// - Data persistence across pod recreations with the same volume
+//
+// The test suite verifies the core functionality needed for both stateless and
+// stateful workloads in Kubernetes when using S3 CSI volumes.
+func InitS3MultiVolumeTestSuite() storageframework.TestSuite {
+	return &s3CSIMultiVolumeTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name: "multivolume",
+			TestPatterns: []storageframework.TestPattern{
+				storageframework.DefaultFsPreprovisionedPV,
+			},
+		},
+	}
+}
+
+// GetTestSuiteInfo returns the test suite information including name and test patterns.
+func (t *s3CSIMultiVolumeTestSuite) GetTestSuiteInfo() storageframework.TestSuiteInfo {
+	return t.tsInfo
+}
+
+// SkipUnsupportedTests allows test suites to skip certain tests based on driver capabilities.
+// For S3 multi-volume scenarios, all tests should be supported, so this is a no-op.
+func (t *s3CSIMultiVolumeTestSuite) SkipUnsupportedTests(_ storageframework.TestDriver, _ storageframework.TestPattern) {
+}
+
+// DefineTests implements the test suite by defining all the test cases for multi-volume
+// scenarios. It creates the necessary volume resources, pods, and validation logic.
+func (t *s3CSIMultiVolumeTestSuite) DefineTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	type local struct {
+		resources []*storageframework.VolumeResource
+		config    *storageframework.PerTestConfig
+		driver    storageframework.TestDriver
+	}
+	var (
+		l local
+	)
+
+	f := framework.NewFrameworkWithCustomTimeouts("multivolume", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityLevel = admissionapi.LevelRestricted
+
+	// Clean up resources after each test
+	cleanup := func(ctx context.Context) {
+		var errs []error
+		for _, resource := range l.resources {
+			errs = append(errs, resource.CleanupResource(ctx))
+		}
+		framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup resource")
+	}
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		l = local{
+			driver: driver,
+		}
+		l.config = driver.PrepareTest(ctx, f)
+		ginkgo.DeferCleanup(cleanup)
+	})
+
+	// Common mount options needed for write access by non-root user
+	mountOptions := []string{
+		fmt.Sprintf("uid=%d", defaultNonRootUser),
+		fmt.Sprintf("gid=%d", defaultNonRootGroup),
+		"allow-other",
+		"debug",
+		"debug-crt",
+	}
+
+	// Test 1: Multiple pods accessing the same volume
+	// This test validates the following configuration:
+	//
+	//   [Pod-1]     [Pod-2]
+	//      \           /
+	//       \         /
+	//        \       /
+	//     [Single S3 Volume]
+	//           |
+	//       [S3 Bucket]
+	//
+	// Validates that multiple pods can simultaneously mount the same volume
+	// and read/write data that's visible to all pods.
+	//
+	// This test verifies:
+	// - RWX access mode functionality with S3 volumes
```

Comment:
It looks like the test does not verify the "X" (execution?) part

Reply:
We can't really do a chmod +x on S3 backed filesystem, its a known limitation.
Let me try to add one with "sh some script.sh". I see the RWX gives false pretenses.
`chmod` will give an `operation not permitted` error

Reply:
I think it's fine, as long as this is consistent with what's expected here.
So you mean we cannot execute any script that we might store in an S3 backend? I guess it should work well if we somehow map the permissions in the object tags...

Reply:
We can using sh, added a test [here](https://github.com/scality/mountpoint-s3-csi-driver/pull/22/commits/62276e2105ece55a1e24ee82683ded827fd2aa0d)
Mountpoint does not implement all the features of a POSIX file system
[Reference](https://github.com/awslabs/mountpoint-s3/blob/05f39827035890ced5e62a3824057293bc955279/doc/SEMANTICS.md?plain=1#L3)

Its not a requirement for us right now.